src/contracts/fungible/data/asset.rs

```diff
@@ -134,6 +144,39 @@ pub struct Asset {
     known_allocations: BTreeMap<bitcoin::OutPoint, Vec<Allocation>>,
 }
 
+impl Asset {
+    /// Create an Asset structure from an sqlite asset table entry.
+    /// This fetches all the other tables for entries associated with the given asset
+    /// and recreates the full Asset structure. This should be used while reading Asset data from
+    /// database table.
+    pub fn from_sql_asset(
+        table_value: &SqlAsset,
```

Comment:
You are correct, this will not reduce number of clones inside the function, so it's only about following convention for `from_*` methods. While the only thing that requires clone are the string values for metafields and the rest is either `Copy` or passed by reference to underlying functions, the strings are anyway the only place where we have clones so far. So let's leave it as is.